#!/usr/bin/env python
PACKAGE = "struct_slam"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("cluster_tolerance", double_t, 0, "", 0.15, 0.000, 10.000)
gen.add("min_cluster_size", int_t, 0, "", 7, 0, 15000)
gen.add("max_cluster_size", int_t, 0, "", 15000, 0, 100000)
gen.add("X_wariai", double_t, 0, "", 749.85, 1, 5000)
gen.add("Y_wariai", double_t, 0, "", 749.85, 1, 5000)
gen.add("X_Y_wariai", double_t, 0, "", 749.85, 1, 5000)
gen.add("window_size", double_t, 0, "", 15, 1, 100)
gen.add("X_pcl", double_t, 0, "", 500, 0.01, 5000)
gen.add("Y_pcl", double_t, 0, "", 500, 0.01, 5000)
gen.add("Z_pcl", double_t, 0, "", 1.0, 0, 1000)
gen.add("contrast_min", double_t, 0, "", 0, 0, 1000)
gen.add("contrast_max", double_t, 0, "", 200, 0, 1000)
gen.add("close_open", double_t, 0, "", 4, 0, 10)
gen.add("open_close", double_t, 0, "", 1, 0, 10)
gen.add("nitika", double_t, 0, "", 100, 0, 255)
gen.add("open", double_t, 0, "", 1, 0, 10)


exit(gen.generate(PACKAGE, "struct_slam", "Depth_pcl"))
